cli/src/main/java/com/devonfw/tools/ide/tool/GlobalToolCommandlet.java

```diff
@@ -0,0 +1,101 @@
+package com.devonfw.tools.ide.tool;
+
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.io.FileAccess;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.process.ProcessErrorHandling;
+import com.devonfw.tools.ide.repo.ToolRepository;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * {@link ToolCommandlet} that are installed globally.
+ */
+public abstract class GlobalToolCommandlet extends ToolCommandlet {
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   * @param tool the {@link #getName() tool name}.
+   * @param tags the {@link #getTags() tags} classifying the tool. Should be created via {@link Set#of(Object) Set.of}
+   *        method.
+   */
+  public GlobalToolCommandlet(IdeContext context, String tool, Set<String> tags) {
+
+    super(context, tool, tags);
+  }
+
+  /**
+   * Override this if the global tool comes with a file that has to be extracted.
+   * @return {@code true} to extract (unpack) the downloaded binary file, {@code false} otherwise.
+   */
+  @Override
+  protected boolean isExtract() {
```

Comment:
Never duplicate the JavaDoc for overridden methods (that is one of the reasons for `@Override` annotation).
The JavaDoc will always be inherited and all IDEs support this and also when you generate the JavaDoc as HTML this will happen. When you copy & paste the JavaDoc and it gets changed, the copy gets outdated and nobody will notice.

You may add additional information to the JavaDoc of inherited methods. But this is only to be done in very few cases.
Please note that JavaDoc describes the API of a method and not its implementation.
If you want to explain the implementation simply use code comments:
```

@Override
protected boolean isExtract() {
  // for global tools we usually download installers and do not want to extract them (e.g. installer.msi file shall not be extracted)
  return false;
}
```

Reply:
Done. Thanks for clarifying

Reply:
I removed `Override it to add custom post installation logic.` from the javaDoc of the method postInstall.